### ISSUE TITLE
Add show user location feature to GeolocateControl

### ISSUE THREAD

**Issue by cusspvz** [mapbox/mapbox-gl-js#4354]
Does `mapbox-gl-js` has a similar api that offers the my location/user location feature available on the native APIs?

```
#android
mapView.setMyLocationEnabled( true )
```

```
#ios
mapView.showsUserLocation = true
```

![location circle](https://www.mapbox.com/android-sdk/images/user-location.png)

**Comment by andrewharvey** [comment]
https://www.mapbox.com/mapbox-gl-js/api/#GeolocateControl

**Comment by cusspvz** [comment]
@andrewharvey thanks for your response. I'm not looking for the geolocation control, but instead, for the user location circle feature.

**Comment by andrewharvey** [comment]
Not out of the box, however you might want to checkout out https://github.com/mapbox/mapbox-gl-js/pull/3781.

Unfortunately the approach I used means it can't be merged into the master branch as is (the user location dot is injected directly into your style which has a bunch of unintended side affects, like if you use setStyle to change the style the dot is lost).

I implemented an alternate version using the HTML Marker, see my recent comments in the thread, but I'm not happy with that approach.

I simply don't have the time right now to make the changes required to build the dot as a custom layer type which works outside of the user visible style, or build it as a plug in. That said I'm happy with how the branch works currently and have been using it in one project.
